src/components/MyTemplate/MyTemplateAlert.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import {
+  AlertDialog,
+  AlertDialogBody,
+  AlertDialogFooter,
+  AlertDialogHeader,
+  AlertDialogContent,
+  AlertDialogOverlay,
+  Button,
+} from "@chakra-ui/react";
+import { useUpdateTemplateMutation, useDeleteTemplateMutation } from "@/redux/api/templateApi";
+
+const MyTemplateAlert = ({ isOpen, cancelRef, onClose, alertHeader, template }: any) => {
+  const isDeleteAlert = alertHeader === "Delete";
+  const [updateTemplate] = useUpdateTemplateMutation();
+  const [deleteTemplate] = useDeleteTemplateMutation();
+
+  const handleDeleteOrUndisplay = () => {
+    if (isDeleteAlert) {
+      deleteTemplate(template._id);
+    } else {
+      template.status = "private";
+      updateTemplate(template);
```

Comment:
When I click _Undisplay_, page does not change, by the way, the type of status stored in backend is enum, so if you change to "private", it may not work

Reply:
I updated backend as well. Could you checkout to my backend branch and try again?